### PR TITLE
integrals/intpoly : Feature to compute integrals for list of polynomials up to max_degree(3D use case)

### PR DESCRIPTION
**Pull request by ArifAhmed1995**
This PR adds the feature of being able to compute integrals of a list of polynomials over a 3-Polytope. Also, if only `max_degree` is supplied then return a dictionary of all monomials up to `max_degree`

Example : 
```
>>> cube1 = [[(0, 0, 0), (0, 6, 6), (6, 6, 6), (3, 6, 0),
...               (0, 6, 0), (6, 0, 6), (3, 0, 0), (0, 0, 6)],
...              [1, 2, 3, 4], [3, 2, 5, 6], [1, 7, 5, 2], [0, 6, 5, 7],
...              [1, 4, 0, 7], [0, 4, 3, 6]]
>>> polytope_integrate(cube1, max_degree=3)
{x*y*z: 4131, 1: 162, x: 378, x**3: 22599/5, x*z: 1377, y**2*z: 6480, z**3: 52488/5, y: 486, x*y**2: 4536, y*z: 1620, z**2: 2268, y**2: 1944, z: 540, x*z**2: 30132/5, y*z**2: 6804, x*y: 1134, x**2*y: 3645, y**3: 8748, x**2*z: 23814/5, x**2: 1215}
>>> polytope_integrate(cube1, [1, x**3, z**3], max_degree=3)
{1: 162, z**3: 52488/5, x**3: 22599/5}
```

### CODE REVIEW

**Comment by ArifAhmed1995** [comment]
@certik Review please. Once this is merged I'll update this PR as well : [#13229](https://github.com/sympy/sympy/pull/13229)

File: sympy/integrals/tests/test_intpoly.py [hunk 1]
```diff
@@ -484,6 +484,16 @@ def test_polytope_integrate():
     assert Abs(polytope_integrate(echidnahedron, expr) - 253.569603474519) <\
     1e-12
 
+    #  Tests for many polynomials with maximum degree given(2D case).
+    assert polytope_integrate(cube2, [x**2, y*z], max_degree=2) == \
+        {y * z: 3125 / 4, x ** 2: 3125 / 3}
```
Comment (certik) [inline sympy/integrals/tests/test_intpoly.py]: 3125 / S(4), ...

File: sympy/integrals/tests/test_intpoly.py [hunk 2]
```diff
@@ -484,6 +484,16 @@ def test_polytope_integrate():
...
+
+    assert polytope_integrate(cube2, max_degree=2) == \
+        {1: 125, x: 625 / 2, x * z: 3125 / 4, y: 625 / 2, y * z: 3125 / 4,
+         z ** 2: 3125 / 3, y ** 2: 3125 / 3, z: 625 / 2, x * y: 3125 / 4,
+         x ** 2: 3125 / 3}
```
Comment (certik) [inline sympy/integrals/tests/test_intpoly.py]: The same here.

**Review by certik — CHANGES_REQUESTED** [review]
Otherwise I think it looks good.

**Comment by ArifAhmed1995** [comment]
@certik Is it fine now ?

File: sympy/integrals/tests/test_intpoly.py [hunk 3]
```diff
@@ -484,6 +484,16 @@ def test_polytope_integrate():
     assert Abs(polytope_integrate(echidnahedron, expr) - 253.569603474519) <\
     1e-12
 
+    #  Tests for many polynomials with maximum degree given(2D case).
+    assert polytope_integrate(cube2, [x**2, y*z], max_degree=2) == \
+        {y * z: 3125 / S(4), x ** 2: 3125 / S(3)}
+
+    assert polytope_integrate(cube2, max_degree=2) == \
+        {1: 125, x: 625 / S(2), x * z: 3125 / S(4), y: 625 / S(2),
+         y * z: 3125 / S(4), z ** 2: 3125 / S(3), y ** 2: 3125 / 3,
```
Comment (certik) [inline sympy/integrals/tests/test_intpoly.py]: This should be `3125 / S(3)`.

**Review by certik — CHANGES_REQUESTED** [review]
You forgot one. Otherwise it looks good.

**Review by certik — APPROVED** [review]
Thanks for fixing the other fractions as well. I have no more comments. +1 to merge if tests pass.

@ArifAhmed1995 please ping me once the tests succeed, so that I can merge it.

**Comment by ArifAhmed1995** [comment]
@certik The tests pass now.

**Pull request merged**